### DISCUSSION TITLE
Binding on Selector Style

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#6743] [Q&A] [answered]
Hi there, 

Just to ask if we can Bind on Selector Style  like this  ? : 

`{Binding PersonViewModel.Users, Source={StaticResource ViewModelLocator}}" `

I want to compare this bind with Static Value to apply style.

`"{x:Static enums:Status.SuperUser}"`

Or I have mistook something ?

Thank in advance

**Comment by maxkatz6** [comment]
First of all, you can bind classname with following syntax:
```
<UserControl Classes.my-class="{Binding BoolProp}" />
```

With that, in your code it probably will be the best/easiest option to have boolean property in the view model and not calculate it in the xaml.
But if you want, you can use MultiBinding to combine two values, but you will also need to write IMultiBindingConverter

**Reply by Tenjim** [reply]
Hi @maxkatz6 ,

I wanted to change style according to enum value. 
Unfortunately, I use many values from my enum for using boolean property from ViewModel to doing a style with this boolean condition.   

What I had in WPF was DataTrigger but I understood that Avalonia don't have this.

P.S : I have already tried AvaloniaBehaviors and it doesn't work in my case

**Reply by workgroupengineering** [reply]
Using https://github.com/wieslawsoltes/AvaloniaBehaviors

**Reply by Tenjim** [reply]
already tried it but not working on the self control. For that reason I'm want to try other ways

**Comment by Tenjim** [comment, marked as answer]
The workaround I found is to use Tag property to contains binding enum value (in string format) and use a style selector with checking Tag property like this 

`<Style Selector="ListBox#lstBox[Tag=SuperUser] ListBoxItem">`

EDIT :

as mentionned @maxkatz6 (Thanks Max) : 

I added many classes on my control ListBox to make two different styles : 

```
<ListBox x:name="lstBox" 
Classes.User="{Binding PersonViewModel.Users, Converter={ToBoolean}, ConverterParameter="{x:Static enums:Status.User}"}"
Classes.SuperUser="{Binding PersonViewModel.Users, Converter={ToBoolean}, ConverterParameter="{x:Static enums:Status.SuperUser}"}">
```

after that apply style with selector from a style :

`<Style Selector="ListBox.User ListBoxItem">`

`<Style Selector="ListBox.SuperUser ListBoxItem">`
